src/test/java/com/rybka/service/export/JsonExportServiceTest.java

```diff
@@ -0,0 +1,107 @@
+package com.rybka.service.export;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.rybka.configuration.ExportProperty;
+import com.rybka.exception.ExportFailureException;
+import com.rybka.model.CurrencyHistory;
+import lombok.SneakyThrows;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Calendar;
+import java.util.List;
+
+import static org.mockito.Mockito.*;
+
+@ExtendWith(SpringExtension.class)
+class JsonExportServiceTest {
+
+    private final ObjectMapper objectMapper = new ObjectMapper();
+    @Mock
+    private ExportProperty exportProperty;
+
+    @Test
+    public void testOnProperExport() {
+
+        // given
+        var testFolder = "src/test/resources/";
+        var testType = "json";
+        var currencyBase = "USD";
+        var currencyTarget = "EUR";
+        var currencyRate = 0.88;
+        var currencyAmount = 4d;
+        var currencyTotal = currencyRate * currencyAmount;
+        var currencyHistory = new CurrencyHistory(currencyBase, currencyTarget, currencyAmount, currencyRate, currencyTotal);
+        currencyHistory.setDate(Calendar.getInstance().getTime());
+        var testList = List.of(currencyHistory);
+        var jsonExportService = new JsonExportService(objectMapper, exportProperty);
+
+        when(exportProperty.getFolder()).thenReturn(testFolder);
+        when(exportProperty.getType()).thenReturn(testType);
+
+        // when
+        jsonExportService.export(testList);
+        Path testFilePath = findSavedFile(testFolder, testType);
+        var actualResult = readSavedFile(testFilePath);
+
+        // then
+        verify(exportProperty).getType();
+        verify(exportProperty).getFolder();
+
+        Assertions.assertEquals(currencyRate, actualResult.getRate());
+
+        removeSavedFile(testFilePath);
+    }
+
+    @Test
+    public void testOnExportException() {
```

Comment:
The same comments as for the CsvExportServiceTest.

Reply:
fixed